### PR TITLE
Convert more Android controls to AppCompat counterparts

### LINKED ISSUE #9323: Convert over additional non appcompat controls to appcompat
### Description

Pull these changes into .NET MAUI

https://github.com/xamarin/Xamarin.Forms/pull/15303
https://github.com/xamarin/Xamarin.Forms/pull/15279

### PR DESCRIPTION
**Pull request by jfversluis**
### Description of Change

Ports the fixes mentioned in #9323 to .NET MAUI.
This changes some public APIs! But we need to have this in order to be able to keep publishing Android apps to the Play Store.

### Issues Fixed

Fixes #9323

### CODE REVIEW

**Review by PureWeen — CHANGES_REQUESTED** [review]
do we need to replace this

[maui/SwitchCellRenderer.cs at 5e8e41a691d0c8b8ca759c182e6ae89ea3d5c3a3 · dotnet/maui (github.com)](https://github.com/dotnet/maui/blob/5e8e41a691d0c8b8ca759c182e6ae89ea3d5c3a3/src/Controls/src/Core/Compatibility/Handlers/ListView/Android/SwitchCellRenderer.cs#L12)

with this

[Xamarin.Forms/SwitchCellAppCompatRenderer.cs at 52755d219662bbfb0c057995d79e9445d767e577 · xamarin/Xamarin.Forms (github.com)](https://github.com/xamarin/Xamarin.Forms/blob/52755d219662bbfb0c057995d79e9445d767e577/Xamarin.Forms.Platform.Android/Cells/SwitchCellAppCompatRenderer.cs)

File: src/Controls/src/Core/Platform/AlertManager/AlertManager.Android.cs [hunk 1]
```diff
@@ -255,7 +256,7 @@ void OnPromptRequested(IView sender, PromptArguments arguments)
 				alertDialog.SetMessage(arguments.Message);
 
 				var frameLayout = new FrameLayout(Activity);
-				var editText = new EditText(Activity) { Hint = arguments.Placeholder, Text = arguments.InitialValue };
+				var editText = new AppCompatEditText(Activity) { Hint = arguments.Placeholder, Text = arguments.InitialValue };
```
Comment (PureWeen) [inline src/Controls/src/Core/Platform/AlertManager/AlertManager.Android.cs:259]: @jfversluis I found this one as well. 

Looks like we need it done in XF as well?
https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.Android/PopupManager.cs#L204

Comment (jfversluis) [reply src/Controls/src/Core/Platform/AlertManager/AlertManager.Android.cs:259]: There is another one around line 193 in the `AlertManager`

**Review by PureWeen — APPROVED** [review]

File: src/Controls/src/Core/Platform/AlertManager/AlertManager.Android.cs (hunk 1 above)
Comment (PureWeen) [reply src/Controls/src/Core/Platform/AlertManager/AlertManager.Android.cs:259]: That's no instantiating a `TextView` it's just extracting it from the Dialog. I inspected the `TextView` and the `TextView` pulled from the dialog is a `MaterialTextView` so we should be fine here

**Pull request merged**